frontend-wasm/tests/expected/add.hir

```diff
@@ -0,0 +1,27 @@
+module noname
```

Comment:
I think we'll want to replace these tests with `filecheck`/`lit` once that tooling is ready, since it will be a lot less fragile to changes, and we can be more precise about what we actually care about in the code that is emitted. I'm generally wary of tests like this which break if even the slightest change to the format/etc. is made. That said, having a few such tests is still useful.

Luckily it should be relatively painless to replace these tests when my changes to `filecheck` are done (hopefully next week), but something to keep in mind as you are writing more tests.

Reply:
Sounds good.
Yes, those are pretty fragile, but the ease of re-generation kinda offsets it. As our formatting stabilizes, I expect them to break only when we change the compiler code that affects the output. I value their fragility when reviewing PRs, though. I want to see how the changes to the compiler affect the output.
Besides their primary function (golden tests, regressions, etc.) I'm using them to peek under the hood at the different stages of the compilation pipeline and to know what code the semantic tests are executing. It's the sort of thing you put in debug prints to figure out what is happening when things are going south. But in this case, it's already available. I also often open wat, hir, and masm files side by side to follow the code changes through the compilation process. The above probably only works on relatively small code sizes, which is true for most smart contracts. As for their precision, instead of dumping the whole module, they can be narrowed to function/block with extra filtering on top.
With all that said, I'm eager to try the lit tests and see how they shape my workflow.

Reply:
Makes sense! I'm certainly in favor of having a mix of tests, including a set like this. We can discuss further once there is some prior art in the repo for `filecheck` so see if that changes anything from your perspective. Having the compiler executable and debugger will provide a lot of the utility you seek as well, particularly once I add some more options for debugging specific passes by emitting only changes/modifications to stdout (if you are familiar with MLIR, it does this, and it's one of the more useful things they've built into the tooling there).

I'd like to have something more akin to MLIR's conversion infrastructure as well, but that's something I'll explore down the road when I have time to clean up/refactor the IR a bit.

Reply:
Great! I've tried to port a simplified MLIR's conversion infra to pliron. I especially loved how the legality of ops kind of drove the conversion process.